app/media_sources/facebook_media_source.rb

```diff
@@ -61,11 +61,34 @@ def initialize(url)
   # @return [Forki::Post]
   def retrieve_facebook_post
     # Unlike Zorki, Forki expects a full URL
-    # begin
-      Forki::Post.lookup(url)
-    # rescue NoMethodError => e
-    #   debugger
-    # end
+    posts = Forki::Post.lookup(url)
+    self.class.create_aws_key_functions_for_posts(posts)
+
+    return posts if Figaro.env.AWS_REGION.blank?
```

Comment:
I think it's fine, if a bit magical, for us to take presence of `AWS_REGION` as the trigger for enabling S3 upload. But rather than baking in this check all over, can we wrap it in a helper or something? I think a `return posts unless enable_transfer_via_s3()` or something would make more inline sense.